### PR TITLE
Implement StatsBoxGrid using getStaticProps with caching

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- ~~Refactors `StatsBoxGrid` component to fetch data using `getServerSideProps`~~
- Extracted logic for each fetch to `@/lib/api/*`
- ~~Convert `getStaticProps` to `getServerSideProps` for `page/index.tsx`~~
- Add fetch functions to `getStaticProps` using `runOnlyOnce` cache wrapper
- Updating data scheme typing and move to `@/lib/types`
- Extract utility helpers to `@/lib/utils/*` and constants to `@/lib/constants`
- Updates `StatsBoxGrid/index.tsx` and associated components:
  - `StatsBoxGrid/GridItem.tsx`
  - `StatsBoxGrid/RangeSelector.tsx`
  - `StatErrorMessage.tsx`
- Update `StatsBoxGrid/useStatsBoxGrid.tsx` hook logic to account for new data fetching approach
- This removes "loading" handlers since data will be fetched on the server; keeps error handling in event the API data is not available. \*
- Implements `Cache-Control` in fetch headers; defaulting to max age of 24 hours, and stale-while-revalidate of 25 hours.

## Related Issue
[Home page add stats box grid](https://www.notion.so/efdn/Home-page-add-stats-box-grid-d4fb69b63c7c40d9883a218dbc3904a0?pvs=4)

\*After discussion, we may want to remove the error handling and rely on fallback data. Can address in separate PR after initial migration

### CODE REVIEW

File: src/components/StatLoadingMessage.tsx
Comment (wackerow) [inline src/components/StatLoadingMessage.tsx:1]: No more "loading" state; removed this component

File: src/components/StatsBoxGrid/GridItem.tsx [hunk 1]
```diff
@@ -1,38 +1,36 @@
...
+const tooltipContent = (metric: StatsBoxMetric) => (
   <div>
     <Translation id="data-provided-by" />{" "}
     <InlineLink to={metric.apiUrl}>{metric.apiProvider}</InlineLink>
   </div>
 )
 
-interface IGridItemProps {
-  metric: Metric
-  dir?: Direction
```
Comment (wackerow) [inline src/components/StatsBoxGrid/GridItem.tsx:26]: Remove `dir`; managed by top-level `html` attribute now

File: src/components/StatsBoxGrid/index.tsx [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+
 import { GridItem } from "./GridItem"
 import { useStatsBoxGrid } from "./useStatsBoxGrid"
 
-export interface IProps {}
-
-const StatsBoxGrid: React.FC<IProps> = () => {
-  const { metrics } = useStatsBoxGrid()
+type StatsBoxGridProps = {
+  data: AllMetricData
```
Comment (wackerow) [inline src/components/StatsBoxGrid/index.tsx:9]: Data loaded in _page_ component now during `getStaticProps`... passed into this component as a prop

File: src/components/StatsBoxGrid/index.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+}
 
+const StatsBoxGrid = ({ data }: StatsBoxGridProps) => {
+  const metrics = useStatsBoxGrid(data)
```
Comment (wackerow) [inline src/components/StatsBoxGrid/index.tsx:13]: Data passed from prop then processed by useStatsBoxGrid hook

File: src/components/StatsBoxGrid/useStatsBoxGrid.tsx [hunk 4]
```diff
@@ -1,259 +1,100 @@
...
-  isLangRightToLeft,
-} from "../../utils/translations"
+import type { AllMetricData, Lang, StatsBoxMetric } from "@/lib/types"
 
-import { RangeSelector } from "./RangeSelector"
-
-export const ranges = ["30d", "90d"] as const
+import { getLocaleForNumberFormat } from "@/lib/utils/translations"
 
-export interface State {
```
Comment (wackerow) [inline src/components/StatsBoxGrid/useStatsBoxGrid.tsx:18]: Extracted (and refactored some) interfaces/types to `types.ts`

File: src/components/StatsBoxGrid/useStatsBoxGrid.tsx [hunk 5]
```diff
@@ -1,259 +1,100 @@
...
-  const [txs, setTxs] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [nodes, setNodes] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
```
Comment (wackerow) [inline src/components/StatsBoxGrid/useStatsBoxGrid.tsx:78]: No need for `useState` on these four anymore.. we have what we need from `data`, we just format the values and return it with the return object

File: src/components/StatsBoxGrid/useStatsBoxGrid.tsx [hunk 6]
```diff
@@ -1,259 +1,100 @@
...
-    }
-
-    const formatNodes = (nodes: number): string => {
-      return new Intl.NumberFormat(localeForStatsBoxNumbers, {
-        minimumSignificantDigits: 3,
-        maximumSignificantDigits: 4,
-      }).format(nodes)
-    }
-
-    const weiToRoundedEther = (wei: number): number => Math.floor(wei * 1e-18)
```
Comment (wackerow) [inline src/components/StatsBoxGrid/useStatsBoxGrid.tsx:123]: extracted to `utils`

File: src/components/StatsBoxGrid/useStatsBoxGrid.tsx [hunk 7]
```diff
@@ -1,259 +1,100 @@
...
-
-    const fetchTotalStaked = async (): Promise<void> => {
```
Comment (wackerow) [inline src/components/StatsBoxGrid/useStatsBoxGrid.tsx:125]: The four data fetching functions were move to `lib/api/*` to be accessed by the _page_ component

File: src/components/StatsBoxGrid/useStatsBoxGrid.tsx [hunk 8]
```diff
@@ -1,259 +1,100 @@
...
-            `api/v1/ethstore/${lookupDay}`,
-            "https://beaconcha.in"
-          )
-          const {
-            data: { effective_balances_sum_wei: sumWei },
-          } = await getData<IFetchEthstoreResponse>(ethstoreDay)
-          const value = weiToRoundedEther(sumWei)
-          data.push({ timestamp, value })
+  const totalEtherStaked =
+    "error" in totalEthStaked
```
Comment (wackerow) [inline src/components/StatsBoxGrid/useStatsBoxGrid.tsx:66]: This PR left in the "error" state handling before we spoke more about it. Shouldn't hurt for now, but separately we can work to remove these as our data fetching becomes more robust on the server

File: src/pages/what-is-ethereum.tsx
Comment (wackerow) [inline src/pages/what-is-ethereum.tsx:1]: Don't worry so much about these changes; my linter removed the unused imports... they're updated again in #235 which is planned to be merged after this one.

**Review by corwintines — APPROVED** [review]

File: src/components/StatsBoxGrid/GridItem.tsx (hunk 1 above)
Comment (nhsz) [reply src/components/StatsBoxGrid/GridItem.tsx:26]: @wackerow should `dir` be removed from [`HomePage`](https://github.com/ethereum/ethereum-org-fork/blob/6b7925a1a5ec7a7b0c07b6e82209a78b4d08c9ac/src/pages/index.tsx#L226C9-L226C12) then?

**Review by nhsz — CHANGES_REQUESTED** [review]
@wackerow Great job. Found a bug with `/ar` locale, ranges buttons should switch side maybe. Left a comment about `dir` usage on homepage

Screen Shot 2023-12-19 at 20 14 15

**Review by corwintines — APPROVED** [review]
Fixes the issues brought up, going to bring this in

**Pull request merged**